Add file publisher plugin

yiglas opened the pull request:
# I am submitting a new Community Plugin

## Repo URL

<!--- Paste a link to your repo here for easy access -->
Link to my plugin: https://github.com/yiglas/obsidian-file-publisher

## Release Checklist
- [ ] I have tested the plugin on
  - [ ]  Windows
  - [x]  macOS
  - [ ]  Linux
  - [ ]  Android _(if applicable)_
  - [ ]  iOS _(if applicable)_
- [ ] My GitHub release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [X] GitHub release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [X] My README.md describes the plugin's purpose and provides clear usage instructions.
- [X] I have read the tips in https://github.com/obsidianmd/obsidian-releases/blob/master/plugin-review.md and have self-reviewed my plugin to avoid these common pitfalls.
- [x] I have added a license in the LICENSE file.
- [x] My project respects and is compatible with the original license of any code from other plugins that I'm using.
      I have given proper attribution to these other projects in my `README.md`.

joethei commented:
[import * as fs from "fs";](https://github.com/yiglas/obsidian-file-publisher/blob/da8a70801ae98a459bc553e9400ea11819bcf922/main.ts#L6)
[import * as path from "path";](https://github.com/yiglas/obsidian-file-publisher/blob/da8a70801ae98a459bc553e9400ea11819bcf922/main.ts#L15-L16)
This will break on mobile, as Node.js is not available there, best to use the Vault API.

[const adapter = this.app.vault.adapter as FileSystemAdapter;](https://github.com/yiglas/obsidian-file-publisher/blob/da8a70801ae98a459bc553e9400ea11819bcf922/main.ts#L35)
This will also break on mobile, as `FileSystemAdapter` is not used there, best to use the Vault API instead of the Adapter API.

[axios.post(url, data, {](https://github.com/yiglas/obsidian-file-publisher/blob/da8a70801ae98a459bc553e9400ea11819bcf922/main.ts#L96-L101)
I would recommend using the `requestUrl` function from the Obsidian API instead, it will handle some things like CORS automatically.

yiglas commented:
@joethei thank you for the review. I've updated the release 

1. Removed fs and path references
2. Removed the usage of the `FileSystemAdapter` 

I wasn't able to replace the usage of axios, the requestUrl function wouldn't post the body of the request to the API server. I am still looking into but would like to release this version if possible.

liamcain commented:
- [rename](https://github.com/yiglas/obsidian-file-publisher/blob/0d79ab63561615add0d4171eae4d193829f6e271/main.ts#L140) you should use [`renameFile`](https://github.com/obsidianmd/obsidian-api/blob/38dd22168d2925086371bfc59e36fd9121527a39/obsidian.d.ts#L1215) instead of `rename` so that links get renamed too.
- [menu.addItem](https://github.com/yiglas/obsidian-file-publisher/blob/0d79ab63561615add0d4171eae4d193829f6e271/main.ts#L52) maybe only add the menu item if the url is configured.
